packages/composite-checkout/src/components/checkout-submit-button.tsx

```diff
@@ -1,14 +1,22 @@
 import { useI18n } from '@wordpress/react-i18n';
 import { cloneElement } from 'react';
 import joinClasses from '../lib/join-classes';
-import { useFormStatus, FormStatus, usePaymentMethod, useProcessPayment } from '../public-api';
+import {
+	useFormStatus,
+	FormStatus,
+	usePaymentMethod,
+	useProcessPayment,
+	PaymentProcessorSubmitData,
+} from '../public-api';
 import CheckoutErrorBoundary from './checkout-error-boundary';
 
 export default function CheckoutSubmitButton( {
+	validateForm,
 	className,
 	disabled,
 	onLoadError,
 }: {
+	validateForm?: () => boolean;
```

Comment:
If we do end up changing these props, we should update the documentation for this package, as well as the unit tests (but see my other comment about possibly not needing this change).

Reply:
Agreed, I will iterate over your suggestions so this might not be required